core/app/models/spree/inventory_unit.rb

```diff
@@ -26,18 +34,19 @@ def order=(_)
     end
 
     validates_presence_of :shipment, :line_item, :variant
+    validates :state, inclusion: { in: [ON_HAND, BACKORDERED, RETURNED, SHIPPED, CANCELED] }
```

Comment:
The states may change for stores. We should allow them to overwrite a method instead.